listening-comp/backend/transcript_fetcher.py

```
@@ -0,0 +1,175 @@
+                'is_generated': transcript.is_generated,
+                'processed_transcript': processed_transcript,
+                'raw_transcript': transcript_data
+            }
+            
+            # Cache the result
+            self.transcript_cache[cache_key] = result
+            
+            return result
+            
+        except (TranscriptsDisabled, NoTranscriptAvailable) as e:
+            raise Exception(f"No transcript available for video {video_id}: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)